swagger/swagger-generator/generator-core/src/main/java/io/servicecomb/swagger/generator/core/utils/ClassUtils.java

```diff
@@ -217,6 +275,20 @@ public static String correctMethodParameterName(String name) {
     }
 
     public static String correctClassName(String name) {
-        return name.replace("-", "_");
+        String parts[] = name.split("\\.");
+        for (int idx = 0; idx < parts.length; idx++) {
+            String part = parts[idx];
+            if (part.isEmpty()) {
+                parts[idx] = "_";
```

Comment:
change string.split(regex) to string.split(regex, -1) to remains empty part
and add test case